src/blocks.c

```diff
@@ -973,9 +1005,29 @@ static void open_new_blocks(cmark_parser *parser, cmark_node **container,
       (*container)->as.code.fence_length = 0;
       (*container)->as.code.fence_offset = 0;
       (*container)->as.code.info = cmark_chunk_literal("");
-
     } else {
-      break;
+      cmark_llist *tmp;
+      cmark_node *new_container = NULL;
+
+      for (tmp = parser->syntax_extensions; tmp; tmp=tmp->next) {
+        cmark_syntax_extension *ext = (cmark_syntax_extension *) tmp->data;
+
+        if (ext->try_opening_block) {
+          const char *input_cstr = cmark_chunk_to_cstr(input);
+
+          new_container = ext->try_opening_block(
+              ext, indented, parser, *container, input_cstr);
```

Comment:
It would be more efficient to pass a `char*` and size pair instead of converting to a C string.